elytron/WFCORE-4360-CredentialStore_Expression_Resolution.adoc

```diff
@@ -0,0 +1,420 @@
+
+TODO - I need to double check if the format should be `ENC:` or `ENC::` the former may be easier to accidentially interpret as a system property with a
+default value.
+
+=== Encryption
+
+To simplify the first iteration of this enhancement the only supported transformation will be `AES/CBC/PKCS5Padding`, this has been selected as one of the
+transformations all JVMs are required to support.  Additionally as the values to be encryopted will be of varying lengths padding is required as the values
+may not fit neatly into the block size for AES encryption.
+
+Additional requirements such as the generation of an initialisation vector will be handled by the underlying `Cipher` implementation, different providers
+may choose to handle this differently so we will not second guess their requirements.
+
+==== Command Line Encryption
+
+It was initially considered if a new top level command should be added to the WildFly Elytron Tool to support the encryption of clear text values, however as
+this enhancement is only making use a single transformation we do not have a need for many additional configuration parameters beyond those required to
+initialise and access the underlying credential store.
+
+The `credential-store` command will be updated with an additional action `encrypt`, this in turn will make use of two parameters:
+
+ * `--alias` - The alias of the `SecretKey` stored within the credential store.
+ * `--clear-text` - The clear text string to be encrypted.
+
+If the `--clear-text` parameter is omitted from the command the user will be prompted to enter the clear text to be encrypted twice to prevent the sensitive
+data from being retained within the shell's command history.
+
+By default the output of the `encrypt` action will just output the token representing the encrypted value, if the parameter `--summary` is also supplied to
+the command the output will also illustrate how this could be represented as an expression.  However the command line tool will not be aware of the management
+model configuration so any expression representation will be for guidance only.
+
+==== Management Operation
+
+The `expression=encryption` resource will also contain an operation `create-expression` the purpose of this operation being to take a clear text value and
+using one of the defined `resolvers` output an expression that can be used elsewhere in the management model.  The operation will just require two parameters:
+
+ * `resolver` - The name of the defined resolver to use to encrypt the data, if the `expression=resolver` resource has the `default-resolver` attribute 
+ defined this parameter can be omitted.
+ * `data` - The clear text value to be encrypted.
```

Comment:
What about `clear-text` instead of `data`? That would correspond with the command line tool option.

Reply:
+1 I will change that,